Лабораторная работа 5/Task3.py

```diff
@@ -0,0 +1,68 @@
+class Book:
+    """ Базовый класс книги. """
+
+    def __init__(self, name: str, author: str):
+        self._name = name
+        self._author = author
+
+    @property
+    def name(self)-> str:
+        return self._name
+        
+    @property
+    def author(self)-> str:
+        return self._author
+
+    def __str__(self):
+        return f"Книга {self._name}. Автор {self._author}"
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(name={self._name!r}, author={self._author!r})"
+
+
+class PaperBook(Book):
+    def __init__(self, name: str, author: str, pages: int):
+        super().__init__(name, author)
+        self.pages = pages
+        
+    @property
+    def pages(self) -> str:
+        return self._pages
+
+    @pages.setter
+    def pages(self, new_pages: str) -> str:
+        if not isinstance(new_pages, int):
+            raise TypeError("Количество страниц должно быть типа int")
+        if new_pages <= 0:
+            raise ValueError("Количество страниц должно быть положительным числом")
+        self._pages = new_pages
+
+    def __str__(self):
+        return f"Бумажная книга {self._name}. Автор {self._author}"
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(name={self._name!r}, author={self._author!r},pages={self.pages!r})"
+
+
+class AudioBook(Book):
+    def __init__(self, name: str, author: str, duration: float):
+        self.duration = duration
+        
+    @property
+    def duration(self) -> float:
+        return self._duration
+
+    @duration.setter
+    def duration(self, new_duration: float) -> float:
+        if not isinstance(new_duration,  float):
+            raise TypeError("Продолжительность должна быть типа float")
+        if new_duration <= 0:
+            raise ValueError("Продолжительность должна быть положительным числом")
+        self._duration = new_duration
```

Comment:
```suggestion
    @duration.setter
    def duration(self, new_duration: float) -> None:
        if not isinstance(new_duration,  (int, float)):
            raise TypeError("Продолжительность должна быть типа float")
        if new_duration <= 0:
            raise ValueError("Продолжительность должна быть положительным числом")
        self._duration = new_duration
```
Значение продолжительности аудио книги может быть и целым и с плавающей запятой, так что в проверку вносим два возможных типа, чтобы код пропускал переменные обоих типов